### ISSUE TITLE
Migrate interpreter to ReactMethod

### ISSUE THREAD

**Issue by terfloth** [Yakindu/statecharts#2412]
The interpreter still makes use of generated react sequences but it should use the new style react method pattern already applied by the code generators.

This issue will require a larger extension of the existing interpreter infrastructure.

**Comment by rherrmannr** [comment]
@terfloth can this be closed?

**Comment by tkutz** [comment]
No, this is still relevant. The interpreter currently works with the "old" sequencer. We should adapt it to the "new" sequencer, that produces react methods, and by now also runCycle, enter, exit methods.
